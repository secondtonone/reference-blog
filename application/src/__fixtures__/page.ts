import {
  PageViewModel
} from '~/view-model';
import fixtureArticle from '~/__fixtures__/article-content';

const page: PageViewModel = {
  id: 1,
  previewImage: '/previewImage.jpg',
  title: 'title',
  author: {
    id: 111,
    name: 'Olga Testova',
    alias: 'olga-testova',
    email: 'example@example.com',
    position: 'Super seo specialist',
    company: 'Yandex',
    bio:
      'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
    photo: 'https://storage.googleapis.com/semblog-rc/testing/user/olga.jpg',
    socials: {
      webSite: 'https://example.com',
      twitter: 'https://twitter.com/hello',
      facebook: '100009766004824',
      instagram: 'hello.hello',
      linkedin: 'chto-chto-942423146',
    },
    isPublic: true
  },
  publishedAt: new Date('2020-01-01').getTime(),
  category: {
    id: 1,
    name: 'name',
    slug: 'slug',
    lang: 'lang',
    children: null
  },
  timeToRead: 1,
  url: 'url',
  template: 'article-base',
  socialImage: 'socialImage',
  preview: 'preview',
  extraImage: 'extra.png',
  altPreviewImage: 'Preview Image',
  altSocialImage: 'Social Image',
  hidePreviewImage: 0,
  editedAt: new Date('2020-02-02').getTime(),
  metaTitle: 'metaTitle',
  metaDescription: 'metaDescription',
  schemaOrg: 'schemaOrg',
  related: [5916, 5915, 5914, 5913],
  translated: null,
  critical: {
    'article-content': {
      id: 1,
      content: fixtureArticle,
      team: 'team',
      galaxy: 'galaxy'
    }
  },
  alternates: [],
  status: 'published',
  tableOfContent: true
};

export default page;
