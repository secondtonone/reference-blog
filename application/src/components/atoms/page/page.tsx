import styled from 'styled-components';

const Page = (props) => (
  <PageStyled data-test="page" {...props} />
);

const PageStyled = styled.div<{ thin?: boolean }>`
  position: relative;
  font-family: 'Inter', Ubuntu, Helvetica, Arial, sans-serif;
  width: 100%;
  height: 100%;
  min-height: ${({ thin }) => (!thin ? '100vh' : '100%')};
  display: flex;
  flex-direction: column;
  will-change: background-color, color;
  min-width: 0;
`;

export default Page;
