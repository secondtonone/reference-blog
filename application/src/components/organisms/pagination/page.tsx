import { IButtonProps } from '@semcore/button';
import ButtonLink from '~/components/atoms/button-link';
import PaginationLink from './pagination-link';

interface Props extends IButtonProps {
  selected?: boolean,
  number: number,
  onPageClick?: (page: number) => void
  dataTest?: string
}

const Page: React.FC<Props> = ({
  number,
  selected = false,
  onPageClick,
  'data-test': dataTest,
  children,
}) => (
  <PaginationLink page={number}>
    {(url) => (
      <ButtonLink
        data-test={dataTest || 'pagination-page'}
        href={url}
        use="secondary"
        selected={selected}
        className="pagination-page"
        style={{ minWidth: 42 }}
        onClick={(e) => {
          if (onPageClick instanceof Function) {
            onPageClick(number);
          }
          e.preventDefault();
        }}
      >
        {children || number}
      </ButtonLink>
    )}
  </PaginationLink>
);

export default Page;
