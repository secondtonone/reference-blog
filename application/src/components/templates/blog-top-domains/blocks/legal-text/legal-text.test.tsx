/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';
import page from '~/__fixtures__/page';
import PageContext from '~/contexts/page-context';
import LegalText from '.';

describe('Match snapshots LegalText', () => {
  it('default', () => {
    const { asFragment } = render(
      <PageContext.Provider value={{ page: { ...page, url: 'blog/sp-500' } }}>
        <LegalText />
      </PageContext.Provider>, {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
