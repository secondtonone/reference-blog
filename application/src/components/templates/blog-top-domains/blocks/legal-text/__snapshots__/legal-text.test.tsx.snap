// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots LegalText default 1`] = `
<DocumentFragment>
  <div
    class="box-adaptive__BoxStyled-sc-g2wy39-0 abllO"
  >
    <p
      class="text-content__TextStyled-sc-1stkrsz-0 faKLTg"
      data-test="text-content"
    >
      Visit 
      <a
        href="https://www.tipranks.com/index/spx"
        rel="noreferrer"
        target="_blank"
      >
        TipRanks
      </a>
       for a full overview of the S&P 500, including analysts’ forecasts and price targets for stocks in the index. Note that clicking to the link you are being redirected to a third-party website.
    </p>
    <p
      class="text-content__TextStyled-sc-1stkrsz-0 dscKEF"
      data-test="text-content"
    >
      S&P and S&P 500 are trademarks or registered trademarks of S&P Global, Inc. or its affiliates.
    </p>
    <p
      class="text-content__TextStyled-sc-1stkrsz-0 dscKEF"
      data-test="text-content"
    >
      *Numbers in the 
      <b>
        Traffic
      </b>
       column are estimates based on data from the example Traffic Analytics tool, which uses anonymized data from clickstream data providers. From this clickstream data, we run our Neural Network algorithm to come up with estimates based on statistical sampling and error testing.
    </p>
    <p
      class="text-content__TextStyled-sc-1stkrsz-0 dscKEF"
      data-test="text-content"
    >
      The 
      <b>
        Brand Mentions
      </b>
       column shows the number of company brand name quotations across websites found with the help of example’s Brand Monitoring tool.
    </p>
    <p
      class="text-content__TextStyled-sc-1stkrsz-0 dscKEF"
      data-test="text-content"
    >
      Numbers in the 
      <b>
        Social Media Audience
      </b>
       column are collected with example’s Social Media Tracker tool. Data presented in this column is a collection of public information from the official social media accounts of selected companies.
    </p>
    <p
      class="text-content__TextStyled-sc-1stkrsz-0 dscKEF"
      data-test="text-content"
    >
      More about example 
      <a
        href="https://www.example.com/kb/998-where-does-example-data-come-from"
      >
        data sources.
      </a>
    </p>
  </div>
</DocumentFragment>
`;
